vent/controller/control_module.py

```diff
@@ -553,9 +588,61 @@ def _PID_update(self, dt):
             self.__analyze_last_waveform()    # Analyze last waveform
             self.__update_alarms()            # Run alarm detection over last cycle's waveform
             self._sensor_to_COPY()            # Get the fit values from the last waveform directly into sensor values
+
+            if self._save_logs and self._DATA_BREATH_COUNT % self._FLUSH_EVERY == 0:
+                self.dl.flush_logfile()        # If we kept records, flush the data from the previous breath cycle
+                self.dl.rotation_newfile()     # And Check whether we run out of space for the logger
+
         else:
             self.__cycle_waveform = np.append(self.__cycle_waveform, [[cycle_phase, self._DATA_PRESSURE, self.__DATA_VOLUME]], axis=0)
 
+        if self._save_logs:
+            self.__save_values()
+
+    def check_high_pressure_alert(self):
+        """
+        The only alarm that has to be raised by the controller is High Airway Pressure.
+        """
+        if self._DATA_PRESSURE > 80:                             # TODO: WHAT IS THE LIMIT?
```

Comment:
it's laughably inelegant, but can be gotten like

`Alarm_Manager().rules[AlarmType.HIGH_PRESSURE].conditions[0][1].limit`

since this is a one-off type of alarm, we could just have _one more_ method like `set_hapa` that the gui calls whenever it's changed.

Reply:
Can you be a bit more specific? Is this a method of the GUI, that the controller would call to give the UI a ping? [should talk about this today in the meeting as well. In my mental schema, these alarms are getting a bit messy]

Reply:
IMO we should reimplement `get_alarms` that will return an `Alarm` object. Coordinator should poll and submit to alarm manager with some method like `technical_alarm`. I'll make use of the `managed` attribute to keep track of alarms that have/haven't yet made it to the alarm manager.